xdsl/dialects/experimental/air.py

```diff
@@ -469,7 +470,7 @@ class HerdTerminatorOp(IRDLOperation):
 class HerdOp(IRDLOperation):
     name = "air.herd"
 
-    sym_name = opt_attr_def(StringAttr)
+    sym_name = opt_prop_def(StringAttr)
```

Comment:
opt_prop_def is a bit broken, I'd recommend against using it, to be honest. The issue is that attributes can be missing in MLIR, but properties cannot. We should probably remove `opt_prop_def` in xDSL